plugins/modules/nmcli.py

```diff
@@ -1924,6 +1944,22 @@ def connection_options(self, detect_change=False):
                     options.update({
                         'wireguard.%s' % name: value,
                     })
+                if self.wireguard.get('wgpeer_public_key'):
+                    options.update({
+                        'wireguard.peer.key': self.wireguard['wgpeer_public_key']
```

Comment:
Why don't you simply call the new option `wireguard.peer.key`, then you don't need this extra code? Same for the others.

Reply:
updated and reverted change to handle values